docs/website/docs/plus/ecosystem/delta.md

```diff
@@ -1,33 +1,191 @@
+
+:::caution
+The `upsert` merge strategy for the Delta destination is **experimental**.
+:::
+
+The `merge` write disposition can be configured as follows on the source/resource level:
+
+<Tabs values={[{"label": "dlt.yml", "value": "yaml"}, {"label": "Python", "value": "python"}]}  groupId="language" defaultValue="yaml">
+  <TabItem value="yaml">
+
+```yaml
+sources:
+  my_source:
+    type: sources.my_source
+    with_args:
+      write_disposition:
+        disposition: merge
+        strategy: upsert
+```
+  </TabItem>
+  <TabItem value="python">
+
 ```py
-from dlt_plus.destinations import delta
+@dlt.resource(
+    primary_key="id",  # merge_key also works; primary_key and merge_key may be used together
+    write_disposition={"disposition": "merge", "strategy": "upsert"},
+)
+def my_resource():
+    yield [
+        {"id": 1, "foo": "foo"},
+        {"id": 2, "foo": "bar"}
+    ]
+...
+
+pipeline = dlt.pipeline("loads_delta", destination="delta")
 
-dest_ = delta(destination_name="filesystem")
 ```
+</TabItem>
+</Tabs>
+
+Or on the `pipeline.run` level: <!-- can this also be defined in the yaml??-->
+
+```py
+pipeline.run(write_disposition={"disposition": "merge", "strategy": "upsert"})
+```
+
+## Partitioning
+
+Delta tables can be partitioned (using [Hive-style partitioning](https://delta.io/blog/pros-cons-hive-style-partionining/)) by specifying one or more partition column hints on the source/resource level:
+
+<Tabs values={[{"label": "dlt.yml", "value": "yaml"}, {"label": "Python", "value": "python"}]}  groupId="language" defaultValue="yaml">
+  <TabItem value="yaml">
+
+  ```yaml
+  sources:
+    my_source:
+      type: sources.my_source
+      with_args:
+        columns:
+          foo:
+            partition: True
+  ```
+
+  </TabItem>
+  <TabItem value="python">
+
+  ```py
+  @dlt.resource(
+    columns={"foo": {"partition": True}}
+  )
+  def my_resource():
+      ...
+
+  pipeline = dlt.pipeline("loads_delta", destination="delta")
+  ```
+
+  </TabItem>
+</Tabs>
+
+:::caution
+Partition evolution (changing partition columns after a table has been created) is not supported.
+:::
+
+## Table access helper functions
+You can use the `get_delta_tables` helper functions to access the native [DeltaTable](https://delta-io.github.io/delta-rs/api/delta_table/) objects. 
+
+```py
+from dlt.common.libs.deltalake import get_delta_tables
+
+...
+
+# get dictionary of DeltaTable objects
+delta_tables = get_delta_tables(pipeline)
+
+# execute operations on DeltaTable objects
+delta_tables["my_delta_table"].optimize.compact()
+delta_tables["another_delta_table"].optimize.z_order(["col_a", "col_b"])
+# delta_tables["my_delta_table"].vacuum()
+# etc.
+```
+
+## Table format
+The Delta destination automatically assigns the `delta` table format to all resources that it will load. You can still fall back to storing files  by setting `table_format` to native on the resource level:
```

Comment:
Hm, what does `native` table format mean?

Reply:
It just means that you're storing the files you're loading as the file format of the original files, instead of delta tables

Reply:
what's an original files here? the files produced from normalization state? So you can select between json, parquet and csv? I'm not sure I understood the concept, maybe it needs additional explanations?